core-api/src/main/java/com/hevelian/identity/core/api/TenantController.java

```diff
@@ -43,10 +50,26 @@ public void deactivateTenant(@Valid @RequestBody TenantDomainDTO tenantDomainDTO
       value = {Default.class, NewTenantGroup.class}) @RequestBody TenantRequestDTO tenant) {
     tenant.getTenantAdmin()
         .setPassword(passwordEncoder.encode(tenant.getTenantAdmin().getPassword()));
-    return new PrimitiveResult<String>(
+    return new PrimitiveResult<>(
         tenantService.addTenant(tenant.toEntity(), tenant.getTenantAdmin().toEntity()).getDomain());
   }
 
+  @RequestMapping(path = "/addTenantLogo", method = RequestMethod.POST)
+  public void addTenantLogo(@ApiParam(value = "Tenant domain", required = true) @RequestParam(name = "tenantDomain") String tenantDomain,
+                            @ApiParam(value = "Tenant logo", required = true) @RequestPart(name = "file") MultipartFile file) throws TenantNotFoundByDomainException,
+      TenantLogoValidator.IllegalLogoException, IOException {
```

Comment:
TenantLogoValidator.IllegalLogoException in unnecessary if you implement validtion using @Valid or @Validated annotations. See the next comment